base/multidimensional.jl

```diff
@@ -85,12 +85,21 @@ immutable CartesianRange{I<:CartesianIndex}
     stop::I
 end
 
+stagedfunction CartesianRange{N}(I::CartesianIndex{N})
+    startargs = fill(1, N)
+    :(CartesianRange(CartesianIndex{$N}($(startargs...)), I))
```

Comment:
You can also use `:(CartesianRange($I($(startargs...)), I))` to call the concrete type directly. This turned out to have some advantage in the performance critical methods like addition, but it probably doesn't matter much here, unless you want to generate many `CartesianRange` objects.